fpsim/wrappers.py

```diff
@@ -0,0 +1,1216 @@
+
+Quick Reference 
+------------------------------
+**What intervention should I model for...**
+
+1. **Better counseling / side effect management** → `set_duration_months()`
+   People stay on methods longer with better support
+   
+2. **Improved product quality** → `set_efficacy()`
+   Better manufacturing or user education increases effectiveness
+   
+3. **Social marketing campaign for specific method** → `set_method_mix()`
+   Shift demand toward targeted methods (requires baseline capture)
+   
+4. **General awareness campaign** → `set_probability_of_use()`
+   Increase overall contraceptive use (CPR)
+   
+5. **Provider training on method transitions** → `scale_switching_matrix()`
+   Make it easier for people to switch to better methods
+   
+6. **Comprehensive program** → Combine multiple methods!
+   Example: Better counseling (duration) + quality improvements (efficacy)
+
+**Method name shortcuts:**
+- 'pill' = Pills
+- 'inj' = Injectables (DMPA, Sayana Press, etc.)
+- 'impl' = Implants (Jadelle, Implanon, etc.)
+- 'iud' = IUDs
+- 'cond' = Male condoms
+- 'btl' = Female sterilization (tubal ligation)
+- 'wdraw' = Withdrawal
+- 'othtrad' = Other traditional methods
+- 'othmod' = Other modern methods
+"""
+# endregion: Introduction
+
+import json
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+from enum import Enum
+from functools import wraps
+from operator import attrgetter
+from pathlib import Path
+from typing import Dict, Optional, Sequence, Union, TypedDict, List, Any
+
+import numpy as np
+import sciris as sc
+
+from . import interventions as fpi
+from .methods import NewMethodConfig
+
+__all__ = [
+    'MethodIntervention',
+    'make_update_methods',
+    'MethodName',
+]
+
+# ==========================================
+# Enums and Type Definitions
+# ==========================================
+class MethodName(str, Enum):
+    """Enumeration of standard contraceptive method names."""
+    NONE = 'none'
+    PILL = 'pill'
+    IUD = 'iud'
+    INJECTABLE = 'inj'
+    CONDOM = 'cond'
+    BTL = 'btl'
+    WITHDRAWAL = 'wdraw'
+    IMPLANT = 'impl'
+    OTHER_TRAD = 'othtrad'
+    OTHER_MOD = 'othmod'
+    
+    @classmethod
+    def values(cls) -> List[str]:
+        """Return list of all method name values."""
+        return [m.value for m in cls]
+    
+    @classmethod
+    def excluding_none(cls) -> List[str]:
+        """Return list of method names excluding 'none'."""
+        return [m.value for m in cls if m != cls.NONE]
+
+class PreviewDict(TypedDict, total=False):
+    """Typed dictionary for preview() return value."""
+    year: float
+    efficacy: Optional[Dict[str, float]]
+    duration_months: Optional[Dict[str, Union[int, float]]]
+    p_use: Optional[float]
+    method_mix: Optional[Dict[str, float]]
+    switching_matrix: Optional[str]
+    new_method: Optional[Dict[str, Any]]
+    label: Optional[str]
+
+@dataclass(slots=True)
+class InterventionConfig:
```

Comment:
Seems like it'd be safe to use dictionaries of inputs in most cases - what's the additional benefit of this class?